03_Data_Management/General.Rmd

```diff
@@ -8,15 +8,47 @@ knitr::include_graphics("images/network/Data Management_General.png")
 
 ### Sort
 
+Sorting out features in ascending or descending order seems a quite primitive operation in any programming language or software package. Even though it is indeed primitive, it is also quite important for filtering and cleaning our datasets. In ArcGIS pro this operation is performed using the tool `Sort`, which is part of the General toolset of the Data Managenent Toolbox. Below we present how we can perform the aforementioned operation using R. For our example we use the Simple Feature object _bezirke_, which depicts the districts within the country of Switzerland Furthermore for performing the sorting operation, we use the column that represents the area (in square km) of every of the districts. 
+
+```{r sort_BaseR}
+library(sf)
+library(arc2r)
+# Read the dataset depicting the districts (Bezirke) in the country of Switzerland
+bezirke <- arc2r::bezirke
+
+# sort the dataset based on the Area in ascending order
+bezirke_asc <- bezirke[order(bezirke$area_km2),]
+head(bezirke_asc)
+
+# sort the dataset based on the Area in descending order
+bezirke_desc <- bezirke[order(-bezirke$area_km2),] 
+head(bezirke_desc)
+
+```
+
+The beauty of R is that offers more than one option to perform a specific operation. In the example above, for performing the sorting operation, we used a simple subsetting method integrated within the so called **base R**. Nevertheless using the the function `arrange()` of the **dpyr package** we are able to produce the exact same result.  
+
+```{r sort_dplyr}
+library(dplyr)
+
+# sort the dataset based on the Area in ascending order
+bezirke_arrange_asc <- arrange(bezirke,area_km2) # by default the function sorts in ascendind order
+head(bezirke_arrange_asc)
+
+# sort the dataset based on the Area in descending order
+bezirke_arrange_desc <- arrange(bezirke,desc(area_km2))
```

Comment:
you could also use `arrange(bezirke, -area_km2))`, right?

Reply:
I could yes

Reply:
...but you dont want to?

Reply:
> 
> 
> ...but you dont want to?

..I do..so I did :)